src/macaron/repo_finder/commit_finder.py

```diff
@@ -341,8 +341,10 @@ def _build_version_pattern(name: str, version: str) -> tuple[Pattern | None, lis
             # - 3.1.test.2 -> 'test' and '2' become optional.
             has_non_numeric_suffix = True
 
-        if count == len(split) - 1 and has_trailing_zero or has_non_numeric_suffix:
-            # This part will be made optional in the regex, hence the grouping bracket.
+        # This part will be made optional in the regex if it matches the correct requirements.
```

Comment:
nit: I think `correct requirements.` in this comment is a bit ambiguous. I think it's also good to put an example to demonstrate the requirements as the case we are handling here is quite specific.